uint32 display_x
uint32 display_y
uint32 distance_z
